pyrsistent/_pmap.py

```diff
@@ -140,7 +140,9 @@ def __str__(self):
 
     def __hash__(self):
         # This hashing algorithm is probably not the speediest
```

Comment:
Yeah, as you mention there are a few comments here and there about caching hashes. The only real reason I haven't done so so far is that it will actually mutate the object. Except for strict purism i cannot really come up with a problem with this. Maybe you can?

If caching is applied I think this comment can be removed. At least for CPython I doubt that looping the elements and doing the operations in python will be any faster than the current implementation.